transmogrifier/sources/ead.py

```diff
@@ -42,10 +95,22 @@ def get_main_titles(cls, xml: Tag) -> list[str]:
         Args:
             xml: A BeautifulSoup Tag representing a single EAD XML record.
         """
-        return [
-            ", ".join(string for string in titleproper.stripped_strings)
-            for titleproper in xml.find_all("titleproper")
-        ]
+        main_titles = []
```

Comment:
Not sure if there's a hole in my logic, but the try/except block seems unnecessary here because `get_optional_fields` runs before `get_required_fields` in `Transformer.transform` so if we try/except for `collection_description.did` there as suggested above, we don't need to duplicate the try/except here given that the record would have errored out already if `collection_description.did` was missing.

Reply:
Yeah, that's true given the current order in which methods are called across the codebase. But this method needs to be fully functional on its own...it has no notion that another check has happened elsewhere in the code that would prevent invalid EAD being passed to it. Setting up a situation where this method only works properly if it's called after another method/function is a fragile architecture that's very likely to cause problems down the road. It's basically the same issue as in my comment on the `generate_name_identifier()` method above, where the code only works if the values passed to the method match our (currently unstated) assumptions.

This is one of the reasons unit tests are important, and now that we're looking at it more closely I think we want to add unit tests for all of these source methods across the repo. I'll add a ticket for doing that in the other sources as maintenance work but we should probably go ahead and add them for this source while we're in the code.

Reply:
Good point, I'll add some unit tests for get_main_titles. I would prefer to do the other methods with that future ticket so I can think more comprehensively about how best to do those across the transforms and because this commit is getting big